src/mctpd.c

```diff
@@ -2760,6 +2828,147 @@ static int bus_endpoint_get_prop(sd_bus *bus,
 	return rc;
 }
 
+static int bus_link_get_prop(sd_bus *bus,
+		const char *path, const char *interface, const char *property,
+		sd_bus_message *reply, void *userdata, sd_bus_error *berr)
+{
+	ctx *ctx = userdata;
+	char *tmpstr = NULL;
+	char *link_name = NULL;
+	link_userdata *lmUserData = NULL;
+	int rc = 0;
+
+	if (!is_interfaces_path(path)) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	rc = sd_bus_path_decode_many(path, MCTP_DBUS_PATH "/%/%", &tmpstr,
+		&link_name);
+	if (rc <= 0) {
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS,
+				"Invalid Object Path");
+		goto out;
+	}
+
+	lmUserData =
+		(link_userdata*)mctp_nl_get_link_userdata_byname(ctx->nl, link_name);
+	if (!lmUserData) {
+		rc = -ENOENT;
+		goto out;
+	}
+	if (lmUserData->role != ENDPOINT_ROLE_UNKNOWN &&
+		lmUserData->role != ENDPOINT_ROLE_ENDPOINT &&
+		lmUserData->role != ENDPOINT_ROLE_BUS_OWNER)
+	{
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS, " %s lmUserData->role %d lmUserData->published %d", __func__, lmUserData->role, lmUserData->published);
+		goto out;
+	}
+
+	if (lmUserData->published != 0 &&
+		lmUserData->published != 1)
+	{
+		sd_bus_error_setf(berr, SD_BUS_ERROR_INVALID_ARGS, " %s lmUserData->role %d lmUserData->published %d", __func__, lmUserData->role, lmUserData->published);
+		goto out;
+	}
```

Comment:
If we want to check whether the userdata is valid, we should include a magic number in the struct, rather than comparing each field against the entire range of valid values.

However, I'm not sure these checks (or a magic nember check) is required anyway.